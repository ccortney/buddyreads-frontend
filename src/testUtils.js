import UserContext from "./auth/UserContext";

const demoUser = {
  id: 0,
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
