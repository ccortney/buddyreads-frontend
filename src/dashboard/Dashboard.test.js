import { render } from "@testing-library/react";
import DashBoard from "./Dashboard";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";


it("renders without crashing", function() {
  render(   
      <MemoryRouter>
        <UserProvider>
            <DashBoard />
        </UserProvider>
      </MemoryRouter>     
  );
});

it("matches snapshot with no buddy reads", function() {
  const { asFragment } = render(      
    <MemoryRouter>
        <UserProvider>
            <DashBoard />
        </UserProvider>
    </MemoryRouter>    );
  expect(asFragment()).toMatchSnapshot();
});
