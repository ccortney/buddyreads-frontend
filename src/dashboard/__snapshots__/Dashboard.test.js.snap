// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no buddy reads 1`] = `
<DocumentFragment>
  <h2>
    Loading...
  </h2>
</DocumentFragment>
`;
